test_programs/execution_success/regression/src/main.nr

```diff
@@ -138,4 +138,32 @@ fn main(x: [u8; 5], z: Field, u: i16, v: i16) {
     assert(-u % -11 == -4);
     assert(u % -11 == u % (v + 2));
     assert(-u % -11 == -u % (v + 2));
+
+    // Issue 6734
+    unsafe {
+        sub_array_extended_into_empty();
+    }
```

Comment:
Can you create a separate directory for this new test? I'd like to avoid `regression` becoming a "kitchen sink" test.